For the following custom error handler, which of the following is wrong?
```js
// Assuming you have Express installed and required it in your code
const express = require('express');

// Create an Express app
const app = express();

// Route handler
app.get('/', (req, res) => {
  // Simulating an error by trying to access a property of an undefined object
  const undefinedObject = undefined;
  const property = undefinedObject.property; // This will throw an error

  // This line will not be executed due to the error above
  res.send('Hello World!');
});

// Default error handler
app.use((err, req, res, next) => {
  // Log the error
  console.error(err);

  // Send an error response to the client
  res.status(500).json({ error: 'Internal Server Error' });
});

// Start the Express app
app.listen(3000, () => {
  console.log('App is running on http://localhost:3000');
});
```

A. The error handler will pass the error to the next error handler in the middleware stack.
B. it is a middleware function that will be called when an error is thrown in the route handler.
C. The error handler will log the error and send a 500 response to the client.
D. The error handler will catch all errors thrown in the route handler including asynchronous errors.

Answer: A